codalab/worker_manager/slurm_batch_worker_manager.py

```diff
@@ -0,0 +1,295 @@
+        # JOBID:NAME (header won't be included with "--noheader" option)
+        # 1487896,john-job-3157358
+        # 1478830,john-job-5234492
+        submitted_jobs = set()
+        jobs = self.run_command(
+            [self.SQUEUE, '-u', self.username, '--format', '%A,%j', '--noheader']
+        )
+        for job in jobs.strip().split():
+            job_id, job_name = job.split(',')
+            if job_name.startswith(self.username) and self.args.job_name in job_name:
+                submitted_jobs.add(job_id)
+        return submitted_jobs
+
+    def get_worker_jobs(self):
+        """
+        Return a list of workers in RUNNING and PENDING state.
+        The current Slurm version on NLP cluster is 17.11.13-2, which doesn't have rest api provided.
+        """
+        # Documentation can be found at https://slurm.schedmd.com
+        # Since allocating resource for a worker may take a while, we periodically check
+        # for worker status and remove those workers that failed at starting phase.
+        jobs_to_remove = set()
+        for job_id in self.submitted_jobs:
+            job_acct = self.run_command([self.SACCT, '-j', job_id, '--format', 'state'])
+            if 'FAILED' in job_acct:
+                jobs_to_remove.add(job_id)
+                logger.error("Failed to start job {}".format(job_id))
+            elif 'COMPLETING' in job_acct or 'COMPLETED' in job_acct or 'CANCELLED' in job_acct:
+                jobs_to_remove.add(job_id)
+        self.submitted_jobs = self.submitted_jobs - jobs_to_remove
+        logger.info("Submitted jobs: ".format(self.submitted_jobs))
+
+        # Get all the Slurm workers that are submitted by SlurmWorkerManager and owned by the current user.
+        # Returning result will be in the following format:
+        # JOBID:STATE (header won't be included with "--noheader" option)
+        # 1478828,PENDING
+        # 1478830,PENDING
+        jobs = self.run_command(
+            [self.SQUEUE, '-u', self.username, '--format', '%A,%T', '--noheader']
+        )
+        jobs = jobs.strip().split()
+        logger.info(
+            'Workers: {}'.format(
+                ' '.join(job for job in jobs if job in self.submitted_jobs) or '(none)'
+            )
+        )
+
+        # Get all the RUNNING jobs that are owned by the current user.
+        # Returning result will be in the following format:
+        # JOBID (header won't be included with "--noheader" option)
+        # 1478828
+        # 1478830
+        running_jobs = self.run_command(
+            [self.SQUEUE, '-u', self.username, '-t', 'RUNNING', '--format', '%A', '--noheader']
+        )
+        running_jobs = running_jobs.strip().split()
+
+        return [
+            WorkerJob(active=True) if job in running_jobs else WorkerJob(active=False)
+            for job in self.submitted_jobs
+        ]
+
+    def start_worker_job(self):
+        """
+        Start a CodaLab Slurm worker that submits batch job to Slurm
```

Comment:
nit, but the codalab slurm workeer doesn't submit a batch job to slurm. maybe `Start a CodaLab Slurm worker by submitting a batch job to slurm`?